include/logger/common.hpp

```diff
@@ -3,24 +3,132 @@
 
 #include <memory>
 #include <string>
+#include <map>
+#include <vector>
 
 namespace logger {
 /**
- * @brief  Log level with their severity order , 0 =>high priority , 5=> least priority.
+ * @brief  Log Levels with their severity order , 0 =>high priority , 5=> least priority.
  */
-enum class LEVEL : int { OFF = 0, FATAL = 1, ERROR = 2, WARN = 3, DEBUG = 4, TRACE = 5 };
+enum class log_level : int { OFF = 0, FATAL = 1, ERROR = 2, WARN = 3, DEBUG = 4, TRACE = 5 };
 
 /**
- * @brief  Interface for the extracted log data like file name , function name etc , one may extend for more
- * parameters such as number of argumnets __VA_ARGS__.
+ *@brief Utility map to store log_levels as key and corresponding value as its equivalent string.
  */
-struct ExtractedLogCredentials
+const std::map<logger::log_level, const std::string> log_level_to_string{ { logger::log_level::OFF, "OFF" },
+  { logger::log_level::FATAL, "FATAL" },
+  { logger::log_level::ERROR, "ERROR" },
+  { logger::log_level::WARN, "WARN" },
+  { logger::log_level::DEBUG, "DEBUG" },
+  { logger::log_level::TRACE, "TRACE" } };
+
+/**
+ * @brief  To deal with extracted log data like file name , function name etc.
+ */
+class ExtractedLogCredentials
```

Comment:
can this be make inaccessible to user as this class is used for internal working of lib